### ISSUE TITLE
 Can you provide the xcode project like unity, or how much can you provide for the xcode project

### ISSUE THREAD

**Issue by matrix3d** [Gamua/Adobe-Runtime-Support#346]

**Comment by zwetan** [comment]
maybe you could bother providing a description of what is the issue exactly?

**Comment by matrix3d** [comment]
> maybe you could bother providing a description of what is the issue exactly?

? Now flexsdk generates ipa directly. Can you provide xcode project and let me generate ipa myself. If you are familiar with unity, you will find that its ipa generation is to generate xcode first, and then generate ipa

**Comment by zwetan** [comment]
>  Can you provide xcode project and let me generate ipa myself.

this never was part of the AIR SDK, so you're asking a new feature then you should descirbe this feature, eg. whatever you want exactly

just a title in an issue does not cover it 

> If you are familiar with unity, you will find that its ipa generation is to generate xcode first, and then generate ipa

that's another SDK, not related to AIR SDK

**Comment by matrix3d** [comment]
Right, now clear the meaning I expressed. I would like to say that this is a new feature, not currently available. If it takes much time and money to develop this function.

**Comment by zwetan** [comment]
OK good luck with that

**Comment by ajwfrost** [comment]
Interesting ask. But you can do this yourself, I would have thought? Creating a standard XCode project and setting this up to generate asssets and launch storyboard would get you part of the way, and then add a custom build action with a script to then call ADT to generate the IPA file.

**Comment by marchbold** [comment]
AIR operates quite differently to Unity in this respect, unity I feel is actually all over the place with it's build processes, macOS it bundles directly skipping the xcode project, Android produces an APK, but iOS/tvOS creates an xcodeproj. 

There is power in the xcode approach especially around adding native code, eg you can just add some objc code and a wrapper to your project and it will get added into the project and compiled with Xcode. But also complications, eg custom entitlements sometimes have to be set manually in the generated project for **each** build, and especially during development, it requires a multiple build process to produce an IPA, unity build c# -> xcodeproj, then Xcode build -> iPA which adds a lot of time to the process. 

Unless you have some real reason for this I would say it isn't needed, AIR has a great build process currently and lots of room for expansion without requiring the xcode project generation.

**Comment by matrix3d** [comment]
没错，as3的打包方式不错。但是我需要xcode工程。我需要对xcode代码进行混淆等一些操作。我现在有一些工具能做到这点。因为为了苹果商店过包。可能这不是一个好的做法。我需要过马甲包


Yes, AS3 is packed in a good way. But I need Xcode engineering. I need to obfuscate the Xcode code. Now I have some tools to do that. Because it's for the apple store. Maybe it's not a good idea. I need a vest

**Comment by JohnBlackburne** [comment]
There are obfuscators for Actionscript, to protect your source code, which I would think is the only code you need to protect. They work on the SWF, so you should be able to add them to your build process a build step after the Actioncript is compiled. If you only need to hide particular features there are custom approaches you can use, such as encryption of strings.

The other code in your app is third party code, in ANEs and in the AIR runtime and libraries. I would think it is up to the writers of that code whether it needs any protection. Generally code protection is much less necessary on iOS as users cannot modify your app, and the scope for copying it to distribute it illegally is limited by Apple's tight control over security.

**Comment by zwetan** [comment]
for iOS any ActionScript bytecode is AOT'ed to native code
it's not like you can get a SWF and decompile it
